include/future_queue.hpp

```diff
@@ -345,8 +345,13 @@ namespace cppext {
     };
 
     /** Internal base class for holding the data which is shared between multiple
-     * instances of the same queue. The base
-     *  class does not depend on the data type and is used by future_queue_base. */
+     * instances of the same queue. The base class does not depend on the data type
+     * and is used by future_queue_base.
+     * Dynamically created instances of this class must only created using new(), but
```

Comment:
Instances of this class must not be created on the stack. They must be dynamically allocated on the heap.
```suggestion
     * Instances of this class must only be created dynamically using new() (don't allocate on the stack!), but
```